### PR TITLE
Retire tpp to loop conversion

### PR DESCRIPTION
**Pull request by chelini**
`linalg-to-xsmm`: is now the default flow. Linalg can either lowers to loop or xsmm.

Changes:

- In `tpp-to-gpu.mlir` for brgemm and gemm the load from memref C is hoisted outside the reduction loop. I updated the tests to more strictly test the IR we generate.

- Replace tpp.brgemm with linalg.batch_reduce_matmul this further reduce dependency from TPP that will be removed soon.

- Replace tpp.gemm with linalg.matmul for the same reason above.

- Replace tpp.add with linalg.add

- Replace tpp.zero with linalg.fill using zero constant.

- Retired `linalg-to-xsmm` flag, as now linalg->xsmm is the default flow.

### CODE REVIEW

**Review by rengolin — COMMENTED** [review]
Looks good to me, thanks! (with a nit).

@adam-smnk, please review the GPU parts and approve the PR once you're happy with it.

File: lib/TPP/DefaultTppPasses.cpp [hunk 1]
```diff
@@ -382,18 +373,11 @@ struct DefaultTppPasses
...
-        pm.addNestedPass<func::FuncOp>(createCleanup());
-      }
-
       // Bufferize: tensor->memref.
       pm.addPass(createBufferize());
 
       // Lower all Tile operations.
-      pm.addNestedPass<func::FuncOp>(
-          createTppLowering(TppLoweringOptions{tppToLoops, linalgToXsmm}));
+      pm.addNestedPass<func::FuncOp>(createTppLowering());
```
[the remark this answers is not in the document]
Comment (chelini) [reply lib/TPP/DefaultTppPasses.cpp:380]: done in a later PR :)

[the remark this answers is not in the document]
Comment (chelini) [reply lib/TPP/DefaultTppPasses.cpp:380]: I can cherry-pick the commit here.

[the remark this answers is not in the document]
Comment (rengolin) [reply lib/TPP/DefaultTppPasses.cpp:380]: Nah, it's ok if it's in a following PR, no worries.

**Review by adam-smnk — APPROVED** [review]
Looks good - all GPU tests pass locally
Some GPU tests might still need a bit of cleanup but I will revisit that later

**Pull request merged**